ipa-core/src/helpers/transport/mod.rs

```diff
@@ -289,15 +289,33 @@ impl RouteParams<RouteId, QueryId, NoStep> for (RouteId, QueryId) {
     }
 }
 
+/// These errors are usually related to an underlying [`Transport::Error`] while broadcasting.
+#[derive(thiserror::Error, Debug)]
+#[error("Error in communicating with peer {peer}: {source}")]
+pub struct BroadcastError<I, E> {
+    pub peer: I,
+    pub source: E,
+}
+
 /// Transport that supports per-query,per-step channels
 #[async_trait]
 pub trait Transport: Clone + Send + Sync + 'static {
     type Identity: TransportIdentity;
     type RecordsStream: BytesStream;
-    type Error: std::fmt::Debug;
+    type Error: std::fmt::Debug + Send;
 
+    /// Return my identity in the network (MPC or Sharded)
     fn identity(&self) -> Self::Identity;
 
+    /// Returns all the identities in this network (MPC or Sharded), myself included.
```

Comment:
curious what is the use case to see all identities, and not just peers? because former can be easily derived on every side

Reply:
It's just a helper function I can use to derive the peers regardless of the generic `Identity`.